reactivesocket/src/test/scala/akka/stream/alpakka/reactivesocket/scaladsl/ReactiveSocketSpec.scala

```diff
@@ -0,0 +1,94 @@
+package akka.stream.alpakka.reactivesocket.scaladsl
+
+import akka.stream.scaladsl._
+import io.reactivesocket.util.PayloadImpl
+import org.scalatest.WordSpec
+
+import scala.concurrent.duration._
+
+class ReactiveSocketSpec extends WordSpec {
+
+  "reactive socket API" should {
+
+    "support server and client" in {
+
+      import io.reactivesocket.transport.tcp.client.TcpTransportClient
+      import io.reactivesocket.transport.tcp.server.TcpTransportServer
+
+      val binding = ReactiveSocketServer(TcpTransportServer.create())
+          .toMat(Sink.foreach {
+            _.flow.runWith(Source.repeat(new PayloadImpl("Pong")), Sink.foreach(println))
```

Comment:
I don't think that is true, you can almost think of them as method types (GET, POST, ...) in http. The server need to know the type to have full information about the request.

I think we should embrace the reactive socket specification instead of hiding/changing it. The different interaction types are defined in the spec for a reason (we like them or not, doesn't matter).